javascript/javascript3/week3/shoppingCart.js

```diff
@@ -0,0 +1,107 @@
+class Product {
+    constructor(name, price) {
+      this.name = name;
+      this.price = price;
+    }
+    convertToCurrency(currency){
+      let price=0;
+      if(currency==="dollars"){
+        price=this.price/6.66;
+      }
+      else if(currency==="rial"){
+        price=this.price*46800;
+      }
+      else if(currency==="euro"){
+        price=this.price/7.5;
+      }else{
+        price=this.price;
+      }
+      return  price;
+    }
+  }
+  
+  class ShoppingCart {
+    constructor() {
+      this.products = [];
+    }
+  
+    addProduct(product) {
+      // Implement functionality here
+      this.products.push(product);
+    }
+  
+    removeProduct(product) {
+      this.products=this.products.filter(function(p){return (p.name!==product.name);});
+      // Implement functionality here
+    }
+  
+    searchProduct(productName) {
+      let foundProducts=this.products.filter(function(p){return p.name===productName;});
+      return foundProducts;
+      // Implement functionality here
+    }
+  
+    getTotal() {
+      let total=0;
+      for(let i=0;i<this.products.length;i++){
+        total+=this.products[i].price;
+      }
+      return total;
+      // Implement functionality here
+    }
+  
+    renderProducts() {
+      let str='';
+      for(let i=0;i<this.products.length;i++){
+        str+='<li>name : ' + this.products[i].name + ' price:' + this.products[i].price + '</li>';
+      }
+      return str;
+      // Implement functionality here
+    }
+  
+    getUser() {
+      const myPromise= new Promise(function(resolve){
```

Comment:
No need to create a new Promise here! Just use async/await! So much cleaner and easier to read.